src/client.js

```diff
@@ -3,14 +3,22 @@ import React from 'react';
 import { hydrate } from 'react-dom';
 import { ClientApp } from './app/containers/App';
 import routes from './app/routes';
+import getRouteProps from './app/routes/getInitialData/utils/getRouteProps';
 
 const data = window.SIMORGH_DATA || {};
+const url = window.location.pathname || '/';
 const root = document.getElementById('root');
 
+const { service } = getRouteProps(routes, url);
+
 hydrate(<ClientApp data={data} routes={routes} />, root);
 
-if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
-  navigator.serviceWorker.register(`/${data.service}/articles/sw.js`);
+if (
+  process.env.NODE_ENV === 'production' &&
+  'serviceWorker' in navigator &&
+  service
```

Comment:
It could match a route, but not have a service in the matched params. We might want to error if there is no service but I think that should probably be handled in `getRouteProps`

Reply:
Can you provide an example of why we might have a route match but no defined service? If that's something we do need to guard against I'd rather we did it in `getRouteProps` instead of here to make this more readable.

Reply:
We might be being a bit overprotective. I think it's just that it is possible we could create a route without a service but wouldn't do so. Happy to remove this if you don't think we need to catch that.

Reply:
I would remove it if we don't have an obvious case where it would need to fail.